buildSrc/src/main/java/net/neoforged/neodev/NeoDevPlugin.java

```diff
@@ -467,6 +449,29 @@ public void apply(Project project) {
         setupProductionServerTest(project, installerJar);
     }
 
+    /**
+     * Get the list of Maven repositories that may contain artifacts for the installer.
+     */
+    private static Provider<List<URI>> getInstallerRepositoryUrls(Project project) {
+        return project.provider(() -> {
+            List<URI> repos = new ArrayList<>();
+            var repositories = project.getRepositories();
+            if (repositories.isEmpty()) {
+                // If no project repos are defined, check dependency management on the settings plugin,
+                // which sadly is inaccessible without internals: https://github.com/gradle/gradle/issues/27260
+                repositories = ((GradleInternal) project.getGradle()).getSettings().getDependencyResolutionManagement().getRepositories();
```

Comment:
Absolutely no reason to use gradle internal stuff here -- this is just for neodev, you can expose that in the settings plugin to the individual projects with a lifecycle beforeProject listener or something without too much trouble, I'd think.